pkg/resources/grant_privileges_to_share.go

```diff
@@ -0,0 +1,474 @@
+				privilegesToRemove,
+				grantOn,
+				sdk.NewAccountObjectIdentifier(id.ShareName.Name()),
+			)
+			if err != nil {
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Error,
+						Summary:  "Failed to revoke removed privileges",
+						Detail:   fmt.Sprintf("Id: %s\nPrivileges to remove: %v\nError: %s", d.Id(), privilegesToRemove, err.Error()),
+					},
+				}
+			}
+		}
+
+		id.Privileges = privilegesAfterChange
+		d.SetId(id.String())
+	}
+
+	return ReadGrantPrivilegesToShare(ctx, d, meta)
+}
+
+func DeleteGrantPrivilegesToShare(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+
+	id, err := ParseGrantPrivilegesToShareId(d.Id())
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to parse internal identifier",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err.Error()),
+			},
+		}
+	}
+
+	err = client.Grants.RevokePrivilegeFromShare(ctx, getObjectPrivilegesFromSchema(d), getShareGrantOn(d), sdk.NewAccountObjectIdentifier(id.ShareName.Name()))
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "An error occurred when revoking privileges from share",
+				Detail:   fmt.Sprintf("Id: %s\nShare name: %s\nError: %s", d.Id(), id.ShareName.FullyQualifiedName(), err.Error()),
+			},
+		}
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func ReadGrantPrivilegesToShare(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	id, err := ParseGrantPrivilegesToShareId(d.Id())
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to parse internal identifier",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err.Error()),
+			},
+		}
+	}
+
+	opts, grantedOn, diags := prepareShowGrantsRequestForShare(id)
+	if len(diags) != 0 {
+		return diags
+	}
+
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	grants, err := client.Grants.Show(ctx, opts)
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to retrieve grants",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err.Error()),
+			},
+		}
+	}
+
+	var privileges []string
+	for _, grant := range grants {
+		if grant.GrantedTo != sdk.ObjectTypeShare {
+			continue
+		}
+		// Only consider privileges that are already present in the ID, so we
+		// don't delete privileges managed by other resources.
+		if !slices.Contains(id.Privileges, grant.Privilege) {
+			continue
+		}
+		if grant.GranteeName.Name() == id.ShareName.Name() { // TODO: id.ShareName should be outside resource identifier (forgot the name)
+			if grantedOn == grant.GrantedOn {
+				privileges = append(privileges, grant.Privilege)
+			}
+		}
+	}
+
+	// It's a pseudo-role, so we have to append it whenever it's specified in the configuration
```

Comment:
i dont know what this means, but it looks like a hack. i guess there's probably a good reason for it though

Reply:
Please describe bit a bit more thoroughly @sfc-gh-jcieslak: 
- attach the privilege name to the comment
- elaborate a bit more on what a pseudo-role is (e.g. it is not returned, etc)
- add a reason for which it has to be added

Reply:
Added more detailed comment